README.md

```diff
@@ -362,86 +362,97 @@ export class Counter {
 
 ### Using Angular 2 Services in your Action Creators
 
-In order to use services in your action creators, you need to integrate
+In order to use services in action creators, we need to integrate
 them into Angular 2's dependency injector.
 
-This means attaching your action creators to a class so that:
+We may as well adopt a more class-based approach to satisfy
+Angular 2's OOP idiom, and to allow us to
 
-1. you can make it `@Injectable()`, and
-2. you can inject other services into its constructor for your
-action creators to use.
+1. make our actions `@Injectable()`, and
+2. inject other services for our action creators to use.
 
-Take a look at this example, which uses
-* [redux-thunk](https://github.com/gaearon/redux-thunk) to
-allow for asynchronous actions, and
-* Angular 2's `http` service to make auth requests.
+Take a look at this example, which injects NgRedux to access
+`dispatch` and `getState` (a replacement for `redux-thunk`),
+and a simple `RandomNumberService` to show a side effect.
 
 ```typescript
 import { Injectable } from '@angular/core';
-import { Http } from '@angular/http';
-
-import {
-  LOGIN_USER_PENDING,
-  LOGIN_USER_SUCCESS,
-  LOGIN_USER_ERROR,
-  LOGOUT_USER
-} from '../constants';
+import { NgRedux } from 'ng2-redux';
+import * as Redux from 'redux';
+import { RootState } from '../store';
+import { RandomNumberService } from '../services/random-number';
 
-// Wrap our action creators in a class and make it @Injectable.
-// Don't forget to add it to your app's `providers`.
 @Injectable()
-export class SessionActions {
-  constructor(private http: Http) {}
+export class CounterActions {
+  constructor (
+    private ngRedux: NgRedux<RootState>,
+    private randomNumberService: RandomNumberService) {}
+
+  static INCREMENT_COUNTER: string = 'INCREMENT_COUNTER';
+  static DECREMENT_COUNTER: string = 'DECREMENT_COUNTER';
+  static RANDOMIZE_COUNTER: string = 'RANDOMIZE_COUNTER';
+
+  // Basic action
+  increment(): void {
+    this.ngRedux.dispatch({ type: CounterActions.INCREMENT_COUNTER });
+  }
 
-  // Here's an action creator that uses HTTP.
-  loginUser(credentials) {
-    return (dispatch, getState) => {
-      dispatch({type: LOGIN_USER_PENDING});
-
-      this.http.post('/auth/login', credentials)
-        .toPromise()
-        .then(response => dispatch({type: LOGIN_USER_SUCCESS, payload: response.json()})
-        .catch(error => dispatch({type: LOGIN_USER_ERROR, payload: error, error: true });
-      });
-    };
+  // Basic action
+  decrement(): void {
+    this.ngRedux.dispatch({ type: CounterActions.DECREMENT_COUNTER });
   }
 
-  // Just a regular, synchronous action creator.
-  logoutUser() {
-    return { type: LOGOUT_USER };
+  // Async action.
+  incrementAsync(delay: number = 1000): void {
+    setTimeout(this.increment.bind(this), delay);
```

Comment:
Not sure if there are drawbacks to using bind, but alternatively we could define increment as an arrow function on the class. This way we can preserve the context of `this`.

I am sure you guys have thought about this. I am just curious why `bind` is a better option?